cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,326 @@
+
+The goal of the scenario is to enable infrastructure operators to configure IRSA support for the AWS provider in Radius to deploy and manage AWS resources.
+
+## Terms and definitions
+
+
+## Objectives
+
+> Issue Reference: https://github.com/radius-project/radius/issues/7618 
+
+### Goals
+
+* Radius users can configure AWS provider and enable Radius to use IRSA for authentication and deployment of application.
+* IRSA can be configured via interactive experience.
+* IRSA can be configured manually.
+* Radius users can deploy and manage AWS resources without having to manage AWS access key id and secret
+
+### Non-goals
+
+* Azure Managed Identity support
+* Azure Workload Identity support
+* Ability of radified applications to be able to use IRSA for authenticating AWS operations.
+
+### User scenarios
+
+As a user, I should be able to configure Radius with IRSA for managing AWS resources.
+
+## Design
+
+### High Level Design for IRSA
+
+During installation, Radius should allow user to enable IRSA for AWS while conifguring AWS provider. Then the user should be able to enter the roleARN associated with desired policies which will be stored as the AWS credential.
+
+At a high level, the full flow to setup IRSA looks as below:
+
+1. configure the cluster with oidc provider. have the role-arn configured with desired policies ready.
+2. take steps to allow  `rad-role-arn` and `eks cluster : namespace: service` to trust each other using aws portal.
+3. use  `rad init` to configure radius with role-arn credential. (or use rad credential later on)
+4. use `rad env update` to configure provider associated with the specific environment (`rad env update qa --aws-account-id <aws-account-id> --aws-region <aws-region>`). 
+5. during rad deploy, radius assumes the role that is configured in the credential. 
+
+Since oidc is configured for a cluster and not specific to a role-arn, the solution can be extended to work in multi-tenancy once we have multi credentials support. 
+
+More details about each step is covered in [Detailed Design](#detailed-design)
+
+IRSA works the same way on EKS as well as non-EKS clusters.  
+
+### Alternatives considered
+
+#### EKS Pod Identity 
+
+EKS Pod Identity was introduced in 2022 as a simplified approach for applications running on EKS to retrieve credentials. It uses the new EKS pod identity instead of OIDC provider for identity. Because of this, users have convenient APIs that allow managing pod identity. 
```

Comment:
> EKS pod identity instead of OIDC provider for identity

Can you explain what's different about these technologies?

I'm trying to square this with your later statement that:

> Radius should be however able to support this feature with minimal code changes, if any.

Reply:
I meant, both approaches allows Radius to assume a AWS role effectively, once the configuration on AWS side is complete. Since we are bringing in support for role-arn as credential, we would not need any code changes.